app/components/files-list.js

```diff
@@ -0,0 +1,55 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { A } from '@ember/array';
+
+export default Component.extend({
+
+  /**
+   * Combine files with any associated actions so they can be displayed
+   */
+  filesWithActions: computed(
+    'files',
+    'submissionActions',
+    'submissionActions.@each.{key,description}',
+    function () {
+      const composition = [];
+
+      const actions = this.get('submissionActions') || A();
+
+      this.get('files').forEach((file) => {
+        const id = file.get('id');
+        const fileActions = actions.filter(a => a.get('key') === id)
+
+        composition.push({
+          file,
+          hasActions: fileActions.length > 0,
+          actions: fileActions
+        });
+      });
+
+      return composition;
+    }
+  ),
+
+  actions: {
+    /**
+     * Bubble the remove request up to parent component if `removeAction` is defined
+     * @param {File} file file obj
+     */
+    doRemove(file) {
+      if (this.removeAction) {
```

Comment:
Interesting, I've never written a guard for a closure action before. Do we expect to handle some other behavior in `doRemove` in the future if `removeAction` is not passed in?

Reply:
`removeAction` may not be passed in (although the button that would have this action shouldn't render in this case anyway). Also, I sometimes write overly defensive code.